subxt/src/rpc/lightclient/background.rs

```diff
@@ -0,0 +1,425 @@
+                if let Err(err) = result {
+                    tracing::warn!(
+                        target: LOG_TARGET,
+                        "Cannot send RPC request to lightclient {:?}",
+                        err.to_string()
+                    );
+                    let (sub_id, _) = self
+                        .id_to_subscription
+                        .remove(&id)
+                        .expect("Channels are inserted above; qed");
+
+                    // Send the error back to frontend.
+                    if sub_id
+                        .send(Err(LightClientError::Request(err.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            "Cannot send RPC request error to id{:?}",
+                            id
+                        );
+                    }
+                }
+            }
+        };
+    }
+
+    /// Parse the response received from the light client and sent it to the appropriate user.
+    async fn handle_rpc_response(&mut self, response: String) {
+        match RpcResponse::from_str(&response) {
+            Ok(RpcResponse::Error { id, error }) => {
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender
+                        .send(Err(LightClientError::Request(error.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                } else if let Some((sub_id_sender, _)) = self.id_to_subscription.remove(&id) {
+                    if sub_id_sender
+                        .send(Err(LightClientError::Request(error.to_string())))
+                        .is_err()
+                    {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                }
+            }
+            Ok(RpcResponse::Method { id, result }) => {
+                // Send the response back.
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender.send(Ok(result)).is_err() {
+                        tracing::warn!(
+                            target: LOG_TARGET,
+                            " Cannot send method response to id {:?}",
+                            id
+                        );
+                    }
+                } else if let Some((sub_id_sender, sender)) = self.id_to_subscription.remove(&id) {
+                    let mut sub_id = result.to_string();
+                    sub_id.retain(|ch| ch.is_ascii_digit());
```

Comment:
The `Box<RawValue>` conversion `to_string()` will have extra quotes around our subscription ID.
This is because we parse the ` subscription_id: "123" ` directly to a `Box<RawValue>` I believe

Reply:
I wonder whether to `trim` the `"`'s then perhaps and then convert `to_string()`? that way if something goes wrong it'll be less likely to be something weird :D